consensus/snapshot/snapshot.h

```diff
@@ -38,6 +40,36 @@ typedef struct snapshot_s {
  */
 retcode_t iota_snapshot_init(snapshot_t *const snapshot, iota_consensus_conf_t *const conf);
 
+/**
+ * Resets a snapshot
+ *
+ * @param snapshot The snapshot
+ * @param conf Consensus configuration
+ *
+ * @return a status code
+ */
+retcode_t iota_snapshot_reset(snapshot_t *const snapshot, iota_consensus_conf_t *const conf);
+
+/**
+ * Initializes a snapshot
+ *
+ * @param snapshot The snapshot
+ * @param conf Consensus configuration
+ *
+ * @return a status code
+ */
+retcode_t iota_snapshot_load_built_in_snapshot(snapshot_t *const snapshot, iota_consensus_conf_t *const conf);
```

Comment:
currently yes, because the calling function is `snapshot_init` which chooses  either to load locally or globally, end we can't init ahead with conf values which are global